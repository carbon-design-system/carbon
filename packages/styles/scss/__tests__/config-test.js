/**
 * Copyright IBM Corp. 2018, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @jest-environment node
 */

'use strict';

const { SassRenderer } = require('@carbon/test-utils/scss');

const { render } = SassRenderer.create(__dirname);

describe('@carbon/styles/scss/config', () => {
  test('Public API', async () => {
    const { get } = await render(`
      @use 'sass:meta';
      @use '../config';
      $_: get('variables', meta.module-variables('config'));
    `);

    // Config only exports variables at the moment
    expect(get('variables').value).toMatchSnapshot();
  });

  test('overrides', async () => {
    const { get } = await render(`
      @use 'sass:meta';
      @use '../config' with (
        $prefix: 'test',
        $css--font-face: false,
        $css--reset: false,
        $css--default-type: false,
      );

      $_: get('config', (
        css--default-type: config.$css--default-type,
        css--font-face: config.$css--font-face,
        css--reset: config.$css--reset,
        prefix: config.$prefix,
      ));
    `);

    expect(get('config').value).toEqual({
      'css--default-type': false,
      'css--font-face': false,
      'css--reset': false,
      prefix: 'test',
    });
  });
});
