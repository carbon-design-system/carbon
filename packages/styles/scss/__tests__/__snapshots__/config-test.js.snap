// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@carbon/styles/scss/config Public API 1`] = `
Object {
  "css--default-type": true,
  "css--font-face": true,
  "css--reset": true,
  "flex-grid-columns": 16,
  "font-display": "swap",
  "font-path": "~@ibm/plex",
  "prefix": "cds",
  "use-flexbox-grid": false,
  "use-google-fonts": false,
}
`;
